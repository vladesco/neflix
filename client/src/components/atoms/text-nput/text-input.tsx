import React from 'react'
import cn from 'classnames'

import { AtomReactComponent } from '../types'
import { textInputStyles } from './text-input.style'

type props = {
    placeholder: string
    value?: string | number
    onChange: (value: string | number) => void
}

export const TextInput: AtomReactComponent<props> = ({
    className,
    onChange,
    ...props
}) => {
    const classes = textInputStyles()
    return (
        <input
            className={cn(classes.input, className)}
            onChange={(event) =>
                onChange(event.target.valueAsNumber || event.target.value)
            }
            {...props}
        />
    )
}
