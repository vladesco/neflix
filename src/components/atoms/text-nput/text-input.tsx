import React from 'react'
import cn from 'classnames'

import { AtomReactComponent } from '../types'
import { textInputStyles } from './text-input.style'

type props = {
    placeHolder: string
    value?: string | number
}

export const TextInput: AtomReactComponent<props> = ({
    placeHolder,
    className,
    value,
}) => {
    const classes = textInputStyles()
    return (
        <input
            className={cn(classes.input, className)}
            value={value}
            placeholder={placeHolder}
        />
    )
}
