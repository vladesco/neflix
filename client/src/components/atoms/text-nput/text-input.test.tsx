import React from 'react'
import { screen, render, fireEvent, waitFor } from '@testing-library/react'
import { TextInput } from './text-input'

describe('text input', () => {
    it('should be rendered', () => {
        const testPlaceholder: string = 'test placeholder'
        const { asFragment } = render(
            <TextInput placeholder={testPlaceholder} onChange={null} />
        )

        expect(asFragment()).toMatchSnapshot()
        expect(screen.getByPlaceholderText(testPlaceholder)).toBeTruthy()
    })

    it('should emit entered value', async () => {
        const handleFn = jest.fn()
        const testPlaceholder: string = 'test placeholder'
        const testInputValue: string = 'test input value'
        render(<TextInput placeholder={testPlaceholder} onChange={handleFn} />)

        fireEvent.input(screen.getByPlaceholderText(testPlaceholder), {
            target: { value: testInputValue },
        })

        await waitFor(() =>
            expect(handleFn).toHaveBeenCalledWith(testInputValue)
        )
    })
})
