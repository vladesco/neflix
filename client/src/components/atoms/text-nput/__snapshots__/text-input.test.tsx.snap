// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text input should be rendered 1`] = `
<DocumentFragment>
  <input
    class="input-0-2-1"
    placeholder="test placeholder"
  />
</DocumentFragment>
`;
